### PR TITLE
NJ Regular Exemption

### LINKED ISSUE #1885: NJ Regular Exemption
$1k for everyone, $2k if married filing jointly

Line 6 – Regular Exemptions
You can claim a $1,000 exemption for yourself and your
spouse/CU partner (if filing a joint return) or your domestic
partner. 
https://www.state.nj.us/treasury/taxation/pdf/current/1040i.pdf#page=8

https://www.state.nj.us/treasury/taxation/njit2.shtml#:~:text=Personal%20Exemptions&text=You%20can%20claim%20a%20%241%2C000,claim%20a%20%241%2C000%20regular%20exemption.

### PR DESCRIPTION
**Pull request by ZiqianCui**
Fixes #1885

### CODE REVIEW

**Comment by leogoldman** [comment]
Thank you Leo (@ZiqianCui) for opening your first PolicyEngine PR!! 🥳🥳

**Comment by ZiqianCui** [comment]
Thank you Leo for helping and guiding me these two days!

**Comment by nikhilwoodruff** [comment]
Awesome to see guys! Looks nice and clean so far.

File: policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.NJ
+
+    def formula(tax_unit, period, parameters):
+        # First get their filing status.
+        filing_status = tax_unit("filing_status", period)
+
+        # Then get the NJ Exemptions part of the parameter tree.
+        p = parameters(period).gov.states.nj.exemptions
```
[the remark this answers is not in the document]
Comment (ZiqianCui) [reply policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py:17]: Got it

File: policyengine_us/parameters/gov/states/nj/tax/exemptions/regular/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+description: NJ provides a regular exemption of $1k for individuals and $2k for couples.
+metadata:
+  unit: currency-USD
+  label: NJ Regular Exemption
+  reference:
+    - title: Instructions for Form NJ-1040
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/exemptions/regular/amount.yaml:6]: Could you add a legal code reference?

Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/exemptions/regular/amount.yaml:6]:
```suggestion
    - title: Instructions for Form NJ-1040 (line 6)
```

File: policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+from policyengine_us.model_api import *
+
+
+class nj_regular_exemption(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "NJ regular exemption"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py:7]: I know we're inconsistent on this, but I'd like to move to spelling states out
```suggestion
    label = "New Jersey regular exemption"
```

File: policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    def formula(tax_unit, period, parameters):
+        # First get their filing status.
+        filing_status = tax_unit("filing_status", period)
+
+        # Then get the NJ Exemptions part of the parameter tree.
+        p = parameters(period).gov.states.nj.tax.exemptions
+
+        # Get their regular exemption amount based on their filing status.
+        regular_exemption_amount = p.regular[filing_status]
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py:20]:
```suggestion
        return p.regular[filing_status]
```

**Review by MaxGhenis — COMMENTED** [review]
Looks good! Just a couple suggestions, and please also run `make format`.

**Review by MaxGhenis — COMMENTED** [review]
Also just to add - I'm not immediately sure what the test failure is about, as the filing_status breakdown looks correct. I'll look more later.

**Review by MaxGhenis — COMMENTED** [review]
Please also add an `income/` folder layer, so `states/nj/tax/income/exemptions`

File: policyengine_us/parameters/gov/states/nj/tax/exemptions/regular/amount.yaml (hunk 2 above)
Comment (leogoldman) [reply policyengine_us/parameters/gov/states/nj/tax/exemptions/regular/amount.yaml:6]: @ZiqianCui I think either of these should be good!

https://law.justia.com/codes/new-jersey/2022/title-54a/section-54a-3-1/

https://casetext.com/statute/new-jersey-statutes/title-54a-new-jersey-gross-income-tax-act/chapter-54a3-personal-exemptions-and-deductions/section-54a3-1-personal-exemptions-and-deductions

File: policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py (hunk 4 above)
Comment (nikhilwoodruff) [reply policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py:20]: Yep - though btw think it should be `p.regular.amount`

File: policyengine_us/parameters/gov/states/nj/tax/exemptions/regular/amount.yaml (hunk 2 above)
Comment (ZiqianCui) [reply policyengine_us/parameters/gov/states/nj/tax/exemptions/regular/amount.yaml:6]: Got it, will follow this format! Thanks for the clarification

Comment (ZiqianCui) [reply policyengine_us/parameters/gov/states/nj/tax/exemptions/regular/amount.yaml:6]: I pasted the second source!

File: policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py (hunk 3 above)
Comment (ZiqianCui) [reply policyengine_us/variables/gov/states/nj/tax/exemptions/nj_regular_exemption.py:7]: Got it

File: policyengine_us/parameters/gov/states/nj/tax/income/exemptions/regular/amount.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+description: NJ provides a regular exemption of $1k for individuals and $2k for couples.
+metadata:
+  unit: currency-USD
+  label: NJ Regular Exemption
+  reference: https://casetext.com/statute/new-jersey-statutes/title-54a-new-jersey-gross-income-tax-act/chapter-54a3-personal-exemptions-and-deductions/section-54a3-1-personal-exemptions-and-deductions
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/income/exemptions/regular/amount.yaml:5]: This has to be a separate indented title/href bullet, like the tax form

**Comment by MaxGhenis** [comment]
If you run `git pull upstream master` and push, that should fix the lingering test failure.

**Pull request merged**